optimade/server/routers/utils.py

```diff
@@ -103,7 +104,10 @@ def handle_response_fields(
     new_results = []
     while results:
         new_entry = results.pop(0).dict(exclude_unset=True, by_alias=True)
-
+        if (
+            "hdf5file_path" in new_entry["attributes"]
+        ):  # TODO perhaps hdf5file is not such a good name and we should be more generic like datalocation
+            path = new_entry["attributes"]["hdf5file_path"]  # TODO use Pathlib
```

Comment:
As above, I would suggest making an implementation-specific mixin class for these parameters and not using them in the API responses. If you want to offer direct file download then the `files` endpoint (when it is finished) would be more appropriate, but this just seems like some book-keeping?

Reply:
The idea was to allow downloading only part of the trajectory. For example only the cartesian_positions and not the velocities. And It also gives the option to download only part of the trajectory or every other frame. This is not possible from the files endpoint. Of course we could also offer the hdf5 file for download with the whole trajectory.

Reply:
I can prepend the variable name with an underscore, Thus defining it as a private field. That way, it does not show up on the info endpoint and in openapi.json. So I do not think it is necessary to make an extra class.

Reply:
It seems that Pydantic overwrites some of pythons default behaviour, which causes problems with handling private(properties whose name starts with an underscore) properties. I have now patched the code to correct for this behaviour. But now this solution is not as elegant as I had hoped it to be, so perhaps I should try a different method in the future.